tests/test_introduction_to_classes.py

```diff
@@ -0,0 +1,197 @@
+    try:
+        apple = Fruit(test_name, test_color, test_flavor, test_poisonous, test_weight)
+        apple.bite_it(input_neg)
+        raise Exception("Expected an exception from test_Fruit_bite_it_negative")
+    except AssertionError as e:
+        assert str(e) == "{} error type, please enter integer".format(input_neg)
+
+
+@pytest.mark.parametrize("input_val, expected", [
+    (True, False),
+    (False, True),
+])
+def test_is_edible(input_val, expected):
+
+    orange = Fruit(test_name, test_color, test_flavor, input_val, test_weight)
+    assert orange.is_edible(input_val) == expected
+
+
+def test_description():
+    test_animal_name = "Cat"
+    test_animal_age = 100
+    hippo = Animal(test_animal_name, test_animal_age)
+    assert hippo.description() == test_animal_name + " " + str(test_animal_age)
+
+
+@pytest.mark.parametrize("test_animal_age_neg", [
+    None,
+    True,
+    False,
+    0,
+    -1,
+    ""
+])
+@pytest.mark.parametrize("test_animal_name_neg", [
+    None,
+    True,
+    False,
+    0
+])
+def test_description_negative(test_animal_name_neg, test_animal_age_neg):
+    try:
+        hippo = Animal(test_animal_name_neg, test_animal_age_neg)
+        hippo.description()
+        raise Exception("Expected an exception from test_description_negative")
+    except AssertionError as e:
+        assert str(e) == "Error type, please enter a valid value age or name"
+
+
+@pytest.mark.parametrize("input_val", [
+    1,
+    12,
+    123,
+])
+def test_display_cart(input_val):
+    customer_test = Customer(input_val)
+    assert customer_test.display_cart() == input_val
+
+
+@pytest.mark.parametrize("input_neg", [
+    None,
+    True,
+    False,
+    0,
+    -1,
+    "",
+    "A"
+])
+def test_display_cart_negative(input_neg):
+    try:
+        customer_test = Customer(input_neg)
+        customer_test.display_cart()
+        raise Exception("Expected an exception from test_display_cart_negative")
+    except AssertionError as e:
+        assert str(e) == "Error type, please enter a valid value"
+
+
+@pytest.mark.parametrize("input_val", [
+    1,
+    12,
+    123
+])
+def test_display_order_history(input_val):
+    order_test = ReturningCustomer(input_val)
+    assert order_test.display_order_history() == "I'm a string that stands in for your order history!"
+
+
+@pytest.mark.parametrize("input_neg", [
+    None,
+    True,
+    False,
+    0,
+    -1,
+    "",
+    "A"
+])
+def test_display_cart_negative(input_neg):
+    try:
+        order_test = ReturningCustomer(input_neg)
+        order_test.display_order_history()
+        raise Exception("Expected an exception from display_order_history")
+    except AssertionError as e:
+        assert str(e) == "Error type, please enter a valid value"
+
+
+@pytest.mark.parametrize("test_angle1, test_angle2, test_angle3, expected", [
+    (0, 0, 180, True),
+    (1, 179, 0, True),
+    (1, 1, 178, True),
+    (59, 62, 59, True),
+    (-180, 180, 180, True),
```

Comment:
ughhh, 180 degrees angle produces a straight line! so you can't make a triangle with 180 degrees for any of its angles.
Mathematical proof:
```python
# total sum of all angles must be 180 degrees.
total_sum = 180
#If you give one angle 180 degrees
first_angle = 180
# then other angles only will have 0 degrees to share meaning that it's just a straight line, no corners:
result = total_degrees - first_angle
print(result)
```

so we need to introduce a value check in the triangle class constructor ( `__init__()` ):
`assert 0< angle < 180`